### ISSUE TITLE
Bug: EmptyStatement and associated comments sometimes omitted

### ISSUE THREAD

**Issue by cspotcode** [benjamn/recast#492]
tl;dr When I append or prepend statements to a script and then reprint, `EmptyStatement` nodes and their associated comments are omitted.

Some old code prefixes IIFEs with a semicolon to avoid weird situations where the previous statement forgets a semicolon.  This creates an `EmptyStatement`.

```
IHopeThisStatementDoesntForgetASemicolon()

// This comment is important
;(function() { doStuff; })();
```

Any leading comments are bound to the `EmptyStatement` node created from the semicolon.  recast is sometimes omitting `EmptyStatement` nodes, including their associated comments.

Can someone confirm that this is indeed a bug?  If it's intended behavior, I'll figure out some sort of workaround.

EDIT: I think I found the culprit.
https://github.com/benjamn/recast/blob/master/lib/printer.js#L2307

It should probably be updated to check for the presence of comments.

## Reproduction

Parse source code with a comment bound to an empty statement `;`.  Append a statement to the script via AST manipulation.  Print the ast.

`ast.program.body` has technically changed, so it's reprinted, and the `EmptyStatement` and associated comment are omitted.

```
const code =
`// This comment is bound to an EmptyStatement
;123;`

const recast = require('recast');
const ast = recast.parse(code, {parser: require('recast/parsers/typescript')});
ast.program.body.push({
    type: 'ExpressionStatement',
    expression: {type: 'Identifier', name: 'appendedIdent'}
});
console.log(
    recast.print(ast).code
);
```
prints:
```
123;
appendedIdent;
```
